### ISSUE TITLE
What is the expected behavior of `readme.md`?

### ISSUE THREAD

**Issue by xsc27** [mkdocs/mkdocs#2846]
According to the docs, [Writing-your-docs:Index-pages](https://www.mkdocs.org/user-guide/writing-your-docs/#index-pages), `README.md` will render into `index.html`. There is no mention of `readme.md`. The implementation reflects this behavior: https://github.com/mkdocs/mkdocs/blob/dc35569ade5e101c8c48b78bbed8708dd0b1b49a/mkdocs/structure/files.py#L151

But the check in [get_files()](https://github.com/mkdocs/mkdocs/blob/dc35569ade5e101c8c48b78bbed8708dd0b1b49a/mkdocs/structure/files.py#L225) ignores casing:
https://github.com/mkdocs/mkdocs/blob/dc35569ade5e101c8c48b78bbed8708dd0b1b49a/mkdocs/structure/files.py#L246-L247

I believe it should be:
```python
if filename == 'README.md' and 'index.md' in filenames:
```

**Comment by oprypin** [comment]
Yes, I think this can be treated as a bug; the lowercasing makes no sense.

Would you like to send out the fix to make it so?

> ```python
> if filename == 'README.md' and 'index.md' in filenames:
> ```

Maybe add a new unittest too? Something around https://github.com/mkdocs/mkdocs/blob/82bd8ba5ff177347d094b095bd623a85aaa4c80f/mkdocs/tests/structure/file_tests.py#L628

**Comment by xsc27** [comment]
on it :)
